src/repo/mod.rs

```diff
@@ -214,7 +214,7 @@ impl<TRepoTypes: RepoTypes> Repo<TRepoTypes> {
                 .await
                 .ok();
 
-            if let Ok(kad_subscription) = rx.await? {
+            if let Ok(Ok(kad_subscription)) = rx.await {
```

Comment:
at least this change feels so mysterious that it feels like it would warrant a comment, nor do I follow how does the callers behaviour relate to this. should we squash this kind of bug class somehow or is it fine to change just this one instance?

Aah, I think the case was that oneshot sender which logged the error was complaining that the OneshotSender couldn't be sent through but I am not seeing how does this change help with that.

Reply:
added a relevant `FIXME`; while we might want to do it the right way :tm: in the future, this PR is just a quick fix that makes the current logic consistent.